### PR TITLE
Several new objects for monitoring + inventory config

### LINKED ISSUE #163: Make the exporter configurable for other environments
1. Removal of all static usage-specific configurations
2. Introduction of a configuration file to tailor the values to your own needs.
3. more coming soon

### PR DESCRIPTION
**Pull request by richardtief**
This PR introduces several new objects for monitoring:

- VCOPS Self Monitoring objects (Suite-API, Analytics, vROps Node, vROPs cluster, etc.)
- SDDC Health objects (vCenter Backup Jobs, vCenter Health Services, etc.)
- VMware distributed virtual switches
- NSX-T logical switches

Similarly, the inventory now also has a configuration file to set some behaviors externally. 

1. configure the resource types to be collected in VCOPS and SDDC
2. configure the query specifications for resource requests (VMs have a different setting)

All adapter and resource classes are now created dynamically, so static classes are no longer needed. 
This led to some changes in the creation and provision of the relationship tree. 

I decided not to differentiate between individual VCOPS/SDDC objects. There is now only one GET that collects and provides them all. The prefixes of the associated collectors are set to `vrops_vcops` and `vrops_sddc` respectively. This makes sense since we only need the predefined alerts. 

fixes #163 #168 #167

### CODE REVIEW

**Review by christopherhans — COMMENTED** [review]
Changes are making sense and are coherent.
The changes in i.e. the BaseCollector shows how well designed the exporter is, by more or less copy-pasting the new collector types ;) 

Just for the purpose of understanding: Are you using `vcops` as wording to indicate, that it is about the vrops self monitoring?

**Comment by richardtief** [comment]
Yes, `vcops` is the vCenter Operations Adapter that manages all the self-monitoring objects in vrops.

**Review by christopherhans — APPROVED** [review]

File: InventoryBuilder.py [hunk 1]
```diff
@@ -196,11 +222,15 @@ def query_inventory_permanent(self):
...
             self.provide_datastores()
             self.provide_vms()
+            self.provide_distributed_vswitches()
             self.provide_nsxt_adapter()
             self.provide_nsxt_mgmt_cluster()
             self.provide_nsxt_mgmt_nodes()
             self.provide_nsxt_mgmt_service()
             self.provide_nsxt_transport_nodes()
+            self.provide_nsxt_logical_switches()
+            self.provide_vcops_self_monitoring_objects()
```
Comment (viennaa) [inline InventoryBuilder.py:232]: why is this called `vcops_self_monitoring_objects` but everywhere else you are refering to `vcops_objects`?

File: InventoryBuilder.py [hunk 2]
```diff
@@ -243,92 +277,177 @@ def create_vcenter_objects(self, vrops, target: str, token: str):
...
+                        logger.debug(f'Collecting NSX-T logical switch: {logical_switch.name}')
+
+        return nsxt_adapter_list
+
+    def create_vcops_objects(self, vrops, target: str, token: str):
+        vcops_adapter_instance, self.response_codes[target]["vcops_adapter"] = \
+            Vrops.get_vcenter_operations_adapter_intance(vrops, target, token)
+        vcops_adapter_instance = vcops_adapter_instance[0]
+        if not vcops_adapter_instance:
+            logger.critical(f'Could not get vcops adapter!')
```
Comment (viennaa) [inline InventoryBuilder.py:408]: Why using an f-string here?

File: InventoryBuilder.py [hunk 3]
```diff
@@ -243,92 +277,177 @@ def create_vcenter_objects(self, vrops, target: str, token: str):
...
+                    dc_object.dvss.append(dvs)
+                    logger.debug(f'Collecting distributed virtual switch: {dvs.name}')
         return vcenter_adapter
 
     def create_nsxt_objects(self, vrops, target: str, token: str):
-        nsxt_adapter, self.response_codes[target]["nsxt_adapter"] = Vrops.get_nsxt_adapter(vrops, target, token)
-        if not nsxt_adapter:
+        nsxt_adapter_list, self.response_codes[target]["nsxt_adapter"] = Vrops.get_nsxt_adapter(vrops, target, token)
+        if not nsxt_adapter_list:
+            logger.critical(f'Could not get any nsxt adapter!')
```
Comment (viennaa) [inline InventoryBuilder.py:340]: Why using an f-string here? Maybe bring in some context which vrops we are looking at?

File: InventoryBuilder.py [hunk 4]
```diff
@@ -243,92 +277,177 @@ def create_vcenter_objects(self, vrops, target: str, token: str):
...
+            Vrops.get_vms(vrops, target, token, [hs.uuid for hs in hosts], vcenter_adapter.uuid,
+                          query_specs=query_specs.get('virtual_machines', {}))
+        distributed_virtual_switchs, self.response_codes[target]["distributed_virtual_switch"] = \
+            Vrops.get_dis_virtual_switch(vrops, target, token, [dc.uuid for dc in datacenter],
+                                         query_specs=query_specs.get('default', {}))
 
+        vcenter_adapter.datacenter = list()
         for dc in datacenter:
-            vcenter_adapter.add_datacenter(dc)
+            vcenter_adapter.datacenter.append(dc)
```
Comment (viennaa) [inline InventoryBuilder.py:303]: Really like the object implementation, way better than before.

**Review by viennaa — COMMENTED** [review]
Thanks for providing the implementation. I really like that the Resourcekinds mappers are gone and the object list which is given back now at the resource gathering section. Also the idea to make InventoryBuilder more configureable is pretty cool, let's enhance this a little further. I've opened #188 for discussion.

Some minor notes to this PR are inline.

**Pull request merged**